azkaban-web-server/src/web/js/azkaban/view/job-edit.js

```diff
@@ -260,4 +259,44 @@ $(function () {
   jobEditView = new azkaban.JobEditView({
     el: $('#job-edit-pane')
   });
+
+  formatJobPropertyValues();
+
 });
+
+
+// Improve readability of long job property values
+function formatJobPropertyValues() {
+  var patternsToFormat = new Map();
+
+  // Format values like:
+  // 1) --param1 value1 --param2 --param3 value3
+  // 2) -param1 value -param2 -param3
+  var programArgument = '-[^\\s]+(\\s[^-][^\\s]*)?';
+  var programArgumentRegex =
+      new RegExp('^' + programArgument + '(\\s' + programArgument + ')*$');
+  patternsToFormat.set(programArgumentRegex, function(str) {
+    return str.replace(/\s-/g, "</br>-");
+  });
+
+  // Format values like: something,something,something
+  patternsToFormat.set(/^[^,\s]+(,[^,\s]+)+$/, function(str) {
+    return str.replace(/,/g, ",</br>");
+  });
+
+  // Format values like: something:something:something
+  patternsToFormat.set(/^[^,\s;]+(:[^,\s;]+){5,}$/, function(str) {
+    return str.replace(/:/g, ":</br>");
+  });
```

Comment:
Since the only different between the 2 regexs is , and : - can you factor it out in a function that takes the delimiting character as input?

Reply:
You beat me to it :)

Reply:
Actually, the "," and ":" are not the only difference.
-the regex for comma separated values use: [^,\s]
-the regex for colon separated values use: [^,\s;] - this is intentional, to avoid splitting values by colon where the real delimiter might be a semi-colon, a comma or a space.
Even if it's safe to add ; to the regex for comma separated values for now I would prefer not to encapsulate this in a function because I feel a variable for the delimiter will make the regular expression harder to understand.